File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let total = brown + yellow;
    for (let w = 1; w <= total; w++) {
        let h = total / w;
        if (Number.isInteger(h)) {
            if (2 * (w + h - 2) === brown) {
                return [Math.max(w, h), Math.min(w, h)];
            }
        }
    }
    return [];
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int total = brown + yellow;
        for (int w = 1; w <= total; w++) {
            if (total % w == 0) {
                int h = total / w;
                if (2 * (w + h - 2) == brown) {
                    return new int[]{Math.max(w, h), Math.min(w, h)};
                }
            }
        }
        return new int[]{};
    }
}
